modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/dnd_table/Table.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import classNames from 'classnames';
+import React, {useContext, useEffect, useRef} from 'react';
+
+import TableContext from './TableContext';
+
+function Table({children, className}) {
+	const {draggingColumnName, isFixed, updateTableWidth} = useContext(
+		TableContext
+	);
+	const dndTableRef = useRef(null);
+
+	useEffect(() => {
+		const tableWidth = dndTableRef.current.getBoundingClientRect().width;
+		updateTableWidth(tableWidth);
+	}, [updateTableWidth]);
```

Comment:
These kind of callbacks where you're doing a measurement might be better off as `useLayoutEffect` (there is at least one other place above where you're doing this).